FEXCore/Source/Interface/Context/Context.h

```diff
@@ -373,5 +377,7 @@ class ContextImpl final : public FEXCore::Context::Context {
   std::shared_mutex CustomIRMutex;
   std::atomic<bool> HasCustomIRHandlers {};
   fextl::unordered_map<uint64_t, std::tuple<CustomIREntrypointHandler, void*, void*>> CustomIRHandlers;
+  IntervalList<uint64_t> ForceTSOValidRanges; // The ranges for which ForceTSOInstructions has populated data
```

Comment:
Why does it need this ranges list in addition to ForceTSOInstructions? Since you're already using an ordered set, it would seem that algorithmic complexity shouldn't be a problem, but I haven't fully looked into the implementation details.

Reply:
Not all regions will have TSO metadata, the default tso setting needs to be used in such cases